peps/pep-0790.rst

```diff
@@ -0,0 +1,400 @@
+via documentation, and via just-in-time feedback from static analysis tools.
+In intermediate classes, we recommend teaching ``.flat_exceptions()`` together
+with the ``.split()`` and ``.subgroup()`` methods, and mentioning
+``.preserve_context()`` as an advanced option to address specific pain points.
+
+Both the API reference and the existing `ExceptionGroup tutorial
+<https://docs.python.org/3/tutorial/errors.html#exception-groups>`_ should
+be updated to demonstrate and explain the new methods. The tutorial should
+include examples of common patterns where ``.flat_exceptions()`` and
+``.preserve_context()`` help simplify error handling logic.  Downstream
+libraries which often use exception groups could include similar docs.
+
+We have also designed lint rules for inclusion in ``flake8-async`` which will
+suggest using ``.flat_exceptions()`` when iterating over ``group.exceptions``
+or re-raising a leaf exception, and suggest using ``.preserve_context()`` when
+re-raising a leaf exception inside an ``except*`` block would override any
+existing context.
+
+
+Rejected Ideas
+==============
+
+Add utility functions instead of methods
+----------------------------------------
+
+Rather than adding methods to exceptions, we could provide utility functions:
+
+.. code-block:: python
+
+    def flat_exceptions(exc):
+        if not isinstance(exc, BaseExceptionGroup):
+            return [exc]
+        # implementation
+
+There are however several reasons to prefer methods: there's no obvious place
+where helper functions should live, they take exactly one argument which must
+be an instance of ``BaseException``, and methods are both more convenient and
+more discoverable.
+
+
+Add ``BaseException.as_group()`` (or group methods)
+---------------------------------------------------
+
+Our survey of ``ExceptionGroup``-related error handling code also observed
+many cases of duplicated logic to handle both a bare exception, and the same
+kind of exception inside a group (often incorrectly, motivating
+``.flat_exceptions()``).
+
+We briefly `proposed <https://github.com/python/cpython/issues/125825>`__
+adding ``.split(...)`` and ``.subgroup(...)`` methods too all exceptions,
+before considering ``.flat_exceptions()`` made me feel this was too clumsy.
+As a cleaner alternative, we sketched out an ``.as_group()`` method:
+
+.. code-block:: python
+
+    def as_group(self):
+        if not isinstance(self, BaseExceptionGroup):
+            return BaseExceptionGroup("", [self])
+        return self
+
+However, applying this method to refactor existing code was a negligible
+improvement over writing the trivial inline version.  We also hope that many
+current uses for such a method will be addressed by ``except*`` as older
+Python versions reach end-of-life.
+
+We recommend documenting a "convert to group" recipe for de-duplicated error
+handling, instead of adding group-related methods to ``BaseException``.
+
+
+Footnotes
+=========
+
+.. [#numbers]
```

Comment:
This isn't referenced in the text and breaks the build:
```
/home/runner/work/peps/peps/peps/pep-0790.rst:350: WARNING: Footnote [#] is not referenced. [ref.footnote]
```
https://github.com/python/peps/actions/runs/14349585387/job/40225640358?pr=4357#step:5:136